### PR TITLE
wolfssl, bump version

### PR DESCRIPTION
**Pull request by Begasus**
1 test fails (testsuite.test), 4 test are ok

### CODE REVIEW

File: dev-libs/wolfssl/wolfssl-3.15.3.recipe [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	cmd:make
+	"
+
+defineDebugInfoPackage wolfssl$secondaryArchSuffix \
+	"$libDir"/libwolfssl.so.$libVersion
+
+BUILD()
+{
+	dos2unix *
+	dos2unix ctaocrypt/src/*
```
Comment (alaviss) [inline dev-libs/wolfssl/wolfssl-3.15.3.recipe]: Shall we move these to `PATCH()`?

File: dev-libs/wolfssl/wolfssl-3.15.3.recipe [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	autoreconf -vfi
+	chmod 755 configure
```
Comment (alaviss) [inline dev-libs/wolfssl/wolfssl-3.15.3.recipe]: Wait, what? Shouldn't `autoreconf` generate an executable `configure`?

File: dev-libs/wolfssl/wolfssl-3.15.3.recipe [hunk 3]
```diff
@@ -0,0 +1,89 @@
+SUMMARY="An embedded SSL implementation"
+DESCRIPTION="The wolfSSL embedded SSL library (formerly CyaSSL) is a \
+lightweight SSL/TLS library written in ANSI C and targeted for embedded, \
+RTOS, and resource-constrained environments - primarily because of its \
+small size, speed, and feature set."
+HOMEPAGE="http://yassl.com"
```
Comment (alaviss) [inline dev-libs/wolfssl/wolfssl-3.15.3.recipe]:
```diff
-HOMEPAGE="http://yassl.com"
+HOMEPAGE="https://www.wolfssl.com/"
```

File: dev-libs/wolfssl/wolfssl-3.15.3.recipe [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+BUILD()
+{
+	dos2unix *
+	dos2unix ctaocrypt/src/*
+	autoreconf -vfi
+	chmod 755 configure
+	runConfigure --omit-dirs binDir ./configure \
+		--bindir="$commandBinDir" \
+		--enable-shared
+	make
```
Comment (alaviss) [inline dev-libs/wolfssl/wolfssl-3.15.3.recipe]:
```diff
-	make
+	make $jobArgs
```

File: dev-libs/wolfssl/wolfssl-3.15.3.recipe (hunk 2 above)
Comment (Begasus) [reply dev-libs/wolfssl/wolfssl-3.15.3.recipe]: Still a leftover from previous version, I'll check

File: dev-libs/wolfssl/wolfssl-3.15.3.recipe [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+lightweight SSL/TLS library written in ANSI C and targeted for embedded, \
+RTOS, and resource-constrained environments - primarily because of its \
+small size, speed, and feature set."
+HOMEPAGE="https://www.wolfssl.com/"
+COPYRIGHT="2006-2018 Sawtooth Consulting Ltd."
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://github.com/wolfSSL/wolfssl/archive/v$portVersion-stable.tar.gz"
+CHECKSUM_SHA256="2b98d9ddac2a3188210de720051bed58f91910ee028a4b0a6dd6fd3c9ddb6fec"
+SOURCE_DIR="wolfssl-$portVersion-stable"
```
Comment (alaviss) [inline dev-libs/wolfssl/wolfssl-3.15.3.recipe]:
```diff
 CHECKSUM_SHA256="2b98d9ddac2a3188210de720051bed58f91910ee028a4b0a6dd6fd3c9ddb6fec"
+SOURCE_FILENAME="wolfssl-$portVersion.tar.gz"
 SOURCE_DIR="wolfssl-$portVersion-stable"
```

File: dev-libs/wolfssl/wolfssl-3.15.3.recipe [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="?x86"
```
Comment (alaviss) [inline dev-libs/wolfssl/wolfssl-3.15.3.recipe]: Has the package been tested on any architecture yet?

File: dev-libs/wolfssl/wolfssl-3.15.3.recipe [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+	cmd:make
+	"
+
+defineDebugInfoPackage wolfssl$secondaryArchSuffix \
+	"$libDir"/libwolfssl.so.$libVersion
+
+PATCH()
+{
+	dos2unix *
+	dos2unix ctaocrypt/src/*
```
Comment (alaviss) [inline dev-libs/wolfssl/wolfssl-3.15.3.recipe]: Is this still necessary? I believe GCC should be able to handle files with dos-style endings.

File: dev-libs/wolfssl/wolfssl-3.15.3.recipe (hunk 6 above)
Comment (Begasus) [reply dev-libs/wolfssl/wolfssl-3.15.3.recipe]: it builds ok on x86 (secondaryArch), need to check x86_64
the configuration files look ok and produce the correct paths (don't know yet how to check this ssl version with another source file though)

File: dev-libs/wolfssl/wolfssl-3.15.3.recipe (hunk 7 above)
Comment (Begasus) [reply dev-libs/wolfssl/wolfssl-3.15.3.recipe]: Seems it didn't, builds ok without :)

**Comment by Begasus** [comment]
Build is ok for x86 (secondaryArch) and x86_64, test fails on one part atm ...
<details>
error    test passed!
base64   test passed!
asn      test passed!
MD5      test passed!
SHA      test passed!
SHA-224  test passed!
SHA-256  test passed!
SHA-384  test passed!
SHA-512  test passed!
SHA-3    test passed!
Hash     test passed!
HMAC-MD5 test passed!
HMAC-SHA test passed!
HMAC-SHA224 test passed!
HMAC-SHA256 test passed!
HMAC-SHA384 test passed!
HMAC-SHA512 test passed!
HMAC-SHA3   test passed!
GMAC     test passed!
Chacha   test passed!
POLY1305 test passed!
ChaCha20-Poly1305 AEAD test passed!
AES      test passed!
AES192   test passed!
AES256   test passed!
AES-GCM  test passed!
RANDOM   test passed!
RSA      test passed!
DH       test passed!
ECC      test passed!
logging  test passed!
mutex    test failed!
 error = -9903
FAIL testsuite/testsuite.test (exit status: 255)
</details>

**Comment by alaviss** [comment]
@Begasus Please open an issue for test failures for easier tracking of them. Thanks!

**Pull request merged**